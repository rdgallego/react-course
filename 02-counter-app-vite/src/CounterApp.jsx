import { useState } from 'react'
import PropTypes from 'prop-types'


export const CounterApp = ({value}) => {
    const [ counter, setCounter ] = useState(value);

    const sumar = () => {
        setCounter(counter + 1);
    }
    const restar = () => {
        setCounter(counter - 1);
    }
    const reiniciar = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={sumar}>+1</button>
            <button onClick={restar}>-1</button>
            <button aria-label="btn-reset" onClick={reiniciar}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}