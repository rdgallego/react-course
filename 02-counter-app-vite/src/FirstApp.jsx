/* const newMessage = () => {
    return 'Roberto'
}; */

import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {
    if(!title){
        throw new Error('Titulo no existe')
    }
    return (
        <>
            <h1 data-testid="test-title"> {title} </h1>
            <p>{subtitle * 2}</p>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    title: 'Hola, Soy Roberto',
}