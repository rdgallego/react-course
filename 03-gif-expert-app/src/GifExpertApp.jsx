import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([ 'One Punch', 'Dragon Ball' ]);
    const onAddCategory = (event) => {
        if(categories.includes(event)) return;
        setCategories([...categories, event])
    }
    return (
        <>
            <h1>GifExpert App</h1>
            <AddCategory onAddCategory={ onAddCategory }/>
            {
                categories.map(category =>
                    <GifGrid key={category} category={category}/>
                )
            }
        </>
    )
}