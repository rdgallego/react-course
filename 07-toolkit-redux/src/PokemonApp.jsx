import { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon/thunks"

export const PokemonApp = () => {
  const dispatch = useDispatch()
  const {isLoading, page, pokemons} = useSelector((state) => state.pokemons)
  useEffect(() => {
    return () => {
      dispatch(getPokemons())
    }
  }, [])
  

  return (
    <>
        <h1>Pokemon App</h1>
        <hr/>
        <span>Loading {isLoading ? 'True' : 'False'}</span>
        <ul>
            {pokemons.map((element) => {
              return <li key={element.name}>{element.name}</li>
            })}
        </ul>
        <button disabled={isLoading} onClick={() => {dispatch(getPokemons(page))}}>Next</button>
    </>
  )
}
