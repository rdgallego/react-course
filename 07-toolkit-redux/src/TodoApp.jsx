import { useState } from 'react'
import { useGetTodosQuery, useGetTodoByIdQuery } from './api/todoApi'

export const TodoApp = () => {
    const [todoId, setTodoId] = useState(1);
    const {data: todo = '', isLoading} = useGetTodoByIdQuery(todoId)

    const nextTodo = () => {
        setTodoId(todoId + 1)
    }

    const previousTodo = () => {
        setTodoId(todoId - 1)
    }
  return (
    <>
        <h1>Todo App - RTK Query</h1>
        <hr />
        <h4>isLoading: {isLoading ? 'true' : 'false'} </h4>
        <pre>{JSON.stringify(todo)}</pre>
        <button onClick={() =>nextTodo()}>
            Next
        </button>
        <button onClick={() => previousTodo()}>
            Previous
        </button>
    </>
  )
}
