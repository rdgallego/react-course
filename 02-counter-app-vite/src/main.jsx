import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
/* import {CounterApp} from './CounterApp' */
import './styles.css'

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp subtitle={123}/>
    </React.StrictMode>
)