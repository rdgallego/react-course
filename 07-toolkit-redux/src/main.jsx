import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.jsx'
import './index.css'
import { store } from './store/store.js'
import { Provider } from 'react-redux'
import { PokemonApp } from './PokemonApp.jsx'
import { TodoApp } from './TodoApp.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
     <Provider store={store}>
      <PokemonApp />
     </Provider>

  </React.StrictMode>,
)
