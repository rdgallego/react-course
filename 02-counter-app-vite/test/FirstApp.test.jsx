/* eslint-disable no-unused-vars */
import { render } from "@testing-library/react"
import { FirstApp } from "../src/FirstApp"

/* eslint-disable no-undef */
describe('Pruebas en FirstApp', () => { 
    test('debe hace match con el snapshot', () => { 
        const subtitle = 123;
        const { container } = render( <FirstApp subtitle={subtitle}/> );
        expect(container).toMatchSnapshot();
     })

     test('debe de mostrar el subtitulo en un h1', () => { 
        const subtitle = 123;
        const { container, getByText, getByTestId } = render( <FirstApp subtitle={subtitle}/> );

        expect(getByText(subtitle*2)).toBeTruthy();

/*         const h1 = container.querySelector('h1');
        expect(h1.innerHTML).toContain('Hola, Soy Roberto'); */

        expect(getByTestId('test-title').innerHTML).toContain('Hola, Soy Roberto')
      })

      test('deberia de mostrar el title mandado por props', () => { 

        const subtitle = 123;
        const title = 'Hola, soy Goku';
        const { getByText } = render( <FirstApp title={title} subtitle={subtitle}/> );

        expect(getByText(title).innerHTML).toContain('Hola, soy Goku');

       })
 })