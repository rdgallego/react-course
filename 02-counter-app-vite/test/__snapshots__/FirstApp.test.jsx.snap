// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp debe hace match con el snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, Soy Roberto
     
  </h1>
  <p>
    246
  </p>
</div>
`;
